### PR TITLE
Inputs & Outputs, using real DM, etc [GOL-32]

### PR DESCRIPTION
**Pull request by jboolean**
This PR primarily adds very basic inputs and outputs to Nodes and the ability to move them by dragging. 
Consequently, it also modifies the static test data model to be more similar to the latest data model designs. 

It also contains various cleanup, refactoring or tangential features that I did along the way. 
I do plan to update the method of dragging IOs so they don't feel like they're locked on train tracks. This PR may be declined if feedback warrants a lot of changes.

### CODE REVIEW

File: examples/public/stylesheets/draw/io.less [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.io {
+  circle {
```
Comment (RobertWarrenGilmore) [inline examples/public/stylesheets/draw/io.less]: Shouldn't this start with `&.input` for consistency with line 7.

File: examples/public/stylesheets/main.less [hunk 2]
```diff
@@ -1,69 +1,26 @@
-@import "../vendor/fluidable/fluidable.less";
```
Comment (RobertWarrenGilmore) [inline examples/public/stylesheets/main.less]: Did we ever conclusively figure out what this was doing?

File: golle.sublime-project [hunk 3]
```diff
@@ -2,7 +2,15 @@
...
+			"path": ".",
+      "folder_exclude_patterns": [
+          "node_modules",
+          "examples/public/vendor",
+          "dist"
+      ],
+      "file_exclude_patterns":[
+        "*.sublime-workspace"
+     ]
 		}
```
Comment (RobertWarrenGilmore) [inline golle.sublime-project]: Super weird indentation.

File: examples/public/stylesheets/main.less (hunk 2 above)
Comment (MorganCabral) [inline examples/public/stylesheets/main.less]: Fluidable is a less grid framework that I was using for my parser demo way back. I think getting rid of this will break my parser stuff from a while back, but that is fine because im planing on changing it significantly anyway.

File: lib/test-data.json [hunk 4]
```diff
@@ -8,45 +8,21 @@
...
         }
+        
       ],
-      "outputs": [
-        {
-          "id": "0-output-d",
-          "label": "Delta"
-        }
-      ]
+      "outputs": []
```
Comment (RobertWarrenGilmore) [inline lib/test-data.json]: Why are we getting rid of the output here? Shouldn't we test both inputs and outputs?

File: lib/test-data.json [hunk 5]
```diff
@@ -8,45 +8,21 @@
...
-          "id": "1-output-g",
-          "label": "Gamma"
-        },
-        {
-          "id": "1-output-d",
-          "label": "Delta"
-        }
-      ]
+      "inputs": [],
+      "outputs": []
```
Comment (RobertWarrenGilmore) [inline lib/test-data.json]: Why are we getting rid of these?

File: lib/viewcomponents/io.jsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    /* it's the responsibility of the Node to position the IO since it's in its coordinate space */
+    render: function() {
+        var isInput = this.props.isInput;
+        var ioClass = isInput ? 'input' : 'output';
+
+        var labelPosition = this._getLabelPositioningData(this.props.labelPosition, this.props.size || ioRadius);
+
+        return (
+           <g
+            className={'io ' + ioClass}
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: There's a react addon that handles className concatenation. If we get any classNames much more complicated than this, I would recommend that we use the addon.

File: lib/viewcomponents/io.jsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+var ioRadius = 4;
+/*
+props:
+- isInput {Boolean}
+- label {String}
+- x {Number}
+- y {Number}
+- size (radius) {Number}
+- labelPosition
+ */
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: There is a way to specify property signatures. http://facebook.github.io/react/docs/reusable-components.html#prop-validation

File: lib/viewcomponents/io.jsx (hunk 6 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: http://facebook.github.io/react/docs/class-name-manipulation.html

File: golle.sublime-project (hunk 3 above)
Comment (jboolean) [inline golle.sublime-project]: You're right, not sure what happened there. Fixing.

File: lib/test-data.json (hunk 4 above)
Comment (jboolean) [inline lib/test-data.json]: Laziness. Readding.

File: lib/test-data.json (hunk 5 above)
Comment (jboolean) [inline lib/test-data.json]: Laziness. Re-added.

File: lib/viewcomponents/io.jsx (hunk 7 above)
Comment (jboolean) [inline lib/viewcomponents/io.jsx]: Thanks, that's great! Adding.

File: lib/viewcomponents/io.jsx (hunk 6 above)
Comment (jboolean) [inline lib/viewcomponents/io.jsx]: Thanks for pointing that out. I'll use it here, though it's not quite right for this situation. Definitely something I can see being helpful in the future.

Comment (jboolean) [inline lib/viewcomponents/io.jsx]: Thanks for pointing that out. I'll use it here, though it's not quite right for this situation. Definitely something I can see being helpful in the future.

File: examples/public/stylesheets/draw/io.less (hunk 1 above)
Comment (jboolean) [inline examples/public/stylesheets/draw/io.less]: It's currently overriding it. I'll do that though.

File: examples/public/stylesheets/main.less (hunk 2 above)
Comment (jboolean) [inline examples/public/stylesheets/main.less]: I actually don't recall deleting this line so I can put it back if you'd like.

Comment (RobertWarrenGilmore) [inline examples/public/stylesheets/main.less]: @MorganCabral pls

File: golle.sublime-project [hunk 8]
```diff
@@ -1,8 +1,16 @@
 {
```
Comment (RobertWarrenGilmore) [inline golle.sublime-project]: This file should not be included in the PR. Make sure that you have the latest .gitignore file from the develop branch.

File: lib/enum/ioLabelPosition.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+var IOLabelPosition = {
+  ABOVE: 'above',
+  BELOW: 'below',
+  LEFT: 'left',
+  RIGHT: 'right'
+
+  // Possible future inclusions: perpendicular, perpendicular-flipped, hidden
+  // angled
```
Comment (RobertWarrenGilmore) [inline lib/enum/ioLabelPosition.js]: I like this idea.

File: lib/main.jsx [hunk 10]
```diff
@@ -19,6 +19,10 @@ var dummyModel = {
     // This method is for prototype purposes only.
     getContents: function() {
         return TEST_DATA;
+    },
+
+    render: function() {
```
Comment (RobertWarrenGilmore) [inline lib/main.jsx]: Why did you add this?

File: lib/test-data.json [hunk 11]
```diff
@@ -8,17 +8,23 @@
       "inputs": [
         {
           "id": "0+input-a",
-          "label": "Alpha"
+          "label": "Alpha",
+          "percentage": 25,
+          "direction": "W"
```
Comment (RobertWarrenGilmore) [inline lib/test-data.json]: Shouldn't this use your enums?

File: lib/test-data.json [hunk 12]
```diff
@@ -57,7 +71,9 @@
       "outputs": [
         {
           "id": "2-output-g",
-          "label": "My output"
+          "label": "My output",
```
Comment (RobertWarrenGilmore) [inline lib/test-data.json]: A small, and probably insignificant note - there is currently no requirement or plan to make labels distinct from ids.

File: lib/utilities/positionUtils.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+   * @param  {CardinalDirection} positionData.direction
+   * @param  {Number} positionData.percentage in range [0, 100]
+   * @return {Object} contains x,y placement with origin in top left in range [0, 1].
+   */
+  cardinalToCartesian: function(positionData) {
+    var amount = positionData.percentage/DATA_MODEL_MULTIPLIER;
+
+    switch(positionData.direction) {
+      case CardinalDirection.NORTH:
+      return {x: amount, y: 0};
```
Comment (RobertWarrenGilmore) [inline lib/utilities/positionUtils.js]: No line breaks?

File: lib/viewcomponents/graph.jsx [hunk 14]
```diff
@@ -27,6 +27,8 @@ var Graph = React.createClass({
                     key={nodeData.id}
                     x={nodeData.x}
                     y={nodeData.y}
+                    inputs={nodeData.inputs}
+                    outputs={nodeData.outputs}
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/graph.jsx]: Shouldn't these be getters, in light of the new data models?

File: lib/utilities/positionUtils.js (hunk 13 above)
Comment (jboolean) [inline lib/utilities/positionUtils.js]: Ok. Seemed succinct to me.

File: lib/viewcomponents/graph.jsx (hunk 14 above)
Comment (jboolean) [inline lib/viewcomponents/graph.jsx]: Yes, a lot of things should change in light of the new data models.

File: lib/viewcomponents/io.jsx [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+    x: React.PropTypes.number.isRequired,
+
+    y: React.PropTypes.number.isRequired,
+
+    size: React.PropTypes.number,
+
+    labelPosition: React.PropTypes.oneOf(Object.keys(LabelPosition).map(function(k) {
+      return LabelPosition[k];
+    }))
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: I don't really understand what you're doing here. I get the oneOf and the map function, but what's up with Object.keys? I've never seen that.

Comment (jboolean) [inline lib/viewcomponents/io.jsx]: I wish there was a object.values (maybe I oughta make one). I'm trying to
see that it's a value in the enum.

File: lib/viewcomponents/io.jsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+         {this.props.label}
+       </text>
+     </g>
+     );
+  },
+
+  _getLabelPositioningData: function(labelPosition, extraMargin) {
+    var offset = {x: 0, y: 0};
+
+    var totalMargin = 4 + extraMargin;
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: Is this 4 supposed to be ioRadius?

File: lib/viewcomponents/io.jsx [hunk 17]
```diff
@@ -0,0 +1,78 @@
+var React = require('react');
+var mouseDownDrag = require('../utilities/mouseDownDrag');
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: You declare this and then never use it.

File: lib/viewcomponents/io.jsx [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+         y={labelPosition.y}
+         textAnchor={labelPosition.textAnchor} >
+         {this.props.label}
+       </text>
+     </g>
+     );
+  },
+
+  _getLabelPositioningData: function(labelPosition, extraMargin) {
+    var offset = {x: 0, y: 0};
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: You declare this and then never use it.

File: lib/viewcomponents/node.jsx [hunk 19]
```diff
@@ -1,23 +1,68 @@
...
+      return ios.map(function(io, id) {
+        var position = this._getIOPosition(io);
+
+        var onIOMove = function(event) {
+          this._onIOMove(event, io);
+        }.bind(this);
+
+        return (
+          <IO id={id} isInput={isInput} model={io} 
+          x={position.x} y={position.y} label={io.label} labelPosition={position.labelPosition}
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/node.jsx]: Weird indentation.

File: lib/viewcomponents/node.jsx [hunk 20]
```diff
@@ -1,23 +1,68 @@
...
                     {this.props.label}
                 </text>
-                {/* inputs and outputs */}
+                {this._getRenderedIOs(this.props.inputs, true)}
+                {this._getRenderedIOs(this.props.outputs, false)}
             </g>
         );
     },
 
+    _getRenderedIOs: function(ios, isInput) {
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/node.jsx]: It took me a while to figure out that these are models. It might make it more clear if the naming of these models were changed throughout this file to have names like ioModels, ioModel, nodeModel. This might be something to do later.

File: test/data_model/test-dataModel.js [hunk 21]
```diff
@@ -2,13 +2,15 @@
...
+var CardinalDirection = require('../../lib/enum/cardinalDirection');
+
 var should = require('should'),
     Port = require('../../lib/model/port.js'),
 	Node = require('../../lib/model/node.js'),
 	Connection = require('../../lib/model/connection.js'),
     CardinalPortPosition = require('../../lib/model/cardinalPortPosition.js')
-	port1 = new Port(1,"out",new CardinalPortPosition("N", 50)),
-	port2 = new Port(2,"in", new CardinalPortPosition("S",60)),
+	port1 = new Port(1,"out",new CardinalPortPosition(CardinalDirection.WEST, 50)),
```
Comment (RobertWarrenGilmore) [inline test/data_model/test-dataModel.js]: spacing

File: golle.sublime-project (hunk 8 above)
Comment (jboolean) [inline golle.sublime-project]: Oh, okay. I thought it was helpful to others.

File: lib/main.jsx (hunk 10 above)
Comment (jboolean) [inline lib/main.jsx]: It's a hack to force a re-render after moving the IOs in lieu of writing a proper data flow.

File: lib/test-data.json (hunk 11 above)
Comment (jboolean) [inline lib/test-data.json]: Impossible. This is a json file, it cannot access javascript variables.

File: lib/viewcomponents/io.jsx (hunk 17 above)
Comment (jboolean) [inline lib/viewcomponents/io.jsx]: √

File: lib/viewcomponents/io.jsx (hunk 18 above)
Comment (jboolean) [inline lib/viewcomponents/io.jsx]: √

File: lib/viewcomponents/io.jsx (hunk 16 above)
Comment (jboolean) [inline lib/viewcomponents/io.jsx]: Adding constant.

File: lib/viewcomponents/node.jsx (hunk 20 above)
Comment (jboolean) [inline lib/viewcomponents/node.jsx]: ok. Fixed.

File: lib/viewcomponents/node.jsx (hunk 19 above)
Comment (jboolean) [inline lib/viewcomponents/node.jsx]: Oh. looks line I didn't fix some of the 4-space indents that were here. Also sublime does not understand how to indent jsx.

File: test/data_model/test-dataModel.js (hunk 21 above)
Comment (jboolean) [inline test/data_model/test-dataModel.js]: This file is out of the scope of this PR.

**Comment by jboolean** [comment]
Noticed that the data model has no label property. Is that not a thing?

File: golle.sublime-project (hunk 8 above)
Comment (RobertWarrenGilmore) [inline golle.sublime-project]: It's easier if each developer keeps his own project configuration, because some of us use weird settings that do things like saving draft copies of our own work.

File: lib/viewcomponents/io.jsx (hunk 15 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/io.jsx]: I still don't understand.

File: lib/viewcomponents/node.jsx (hunk 19 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/node.jsx]: You can get a sublime plugin to support jsx.

Comment (jboolean) [inline lib/viewcomponents/node.jsx]: I have. It indents flat though.

File: .gitignore [hunk 22]
```diff
@@ -102,4 +102,6 @@ jison/*.js
 
 # Ignore Sublime Text project files.
 *.sublime-workspace
+*.sublime-project
```
Comment (RobertWarrenGilmore) [inline .gitignore]: You don't have to add this. It's already in develop, on a different line.

Comment (jboolean) [inline .gitignore]: Can't find it.

File: golle.sublime-project (hunk 8 above)
Comment (jboolean) [inline golle.sublime-project]: Fixed.

File: lib/test-data.json (hunk 12 above)
Comment (jboolean) [inline lib/test-data.json]: Oh ok.

File: lib/viewcomponents/io.jsx (hunk 15 above)
Comment (jboolean) [inline lib/viewcomponents/io.jsx]: See new code.

File: lib/model/graphParser.js [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+  if (Object.keys(map).indexOf(id) !== -1) {
+    console.error('Duplicate entity id: ' + id);
+    return false;
+  }
+  map[id] = obj;
+  return true;
+};
+
+var GraphParser = function() {
+  this._nodes = {};
```
Comment (jboolean) [inline lib/model/graphParser.js]: Actually. IDs are global right? The idea here was to 
- check for duplicate ids
- turn IDs into object references, like in the case of edges.

If IDs are global, there's only need for one id -> object mapping.

File: lib/utilities/positionUtils.js [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+      throw 'Unsupported cardinal direction: ' + positionData.getDirection(); 
+    }
+
+  },
+
+  /**
+   * Convert x, y positions between 0 and 1 (fraction of some bound such as a Node) to Cardinal
+   * @method cartesianToCardinal
+   * @param  {CartesianPortPosition} positionData
+   * @return {CartesianPortPosition}
```
Comment (jboolean) [inline lib/utilities/positionUtils.js]: Fixed

File: .gitignore (hunk 22 above)
Comment (RobertWarrenGilmore) [inline .gitignore]: I derped. You're right.

File: lib/model/graphParser.js (hunk 23 above)
Comment (RobertWarrenGilmore) [inline lib/model/graphParser.js]: IDs are supposed to be scoped (once we have containers), and thus possible to duplicate. But the full address of a node (the combination of its ID and its containers' IDs) should be unique.

Comment (RobertWarrenGilmore) [inline lib/model/graphParser.js]: So, for instance, there might be a "Container1>Alpha" and a "Container2>Alpha".

Comment (jboolean) [inline lib/model/graphParser.js]: Ah. Will fix.

Comment (jboolean) [inline lib/model/graphParser.js]: Is that syntax still correct?

Comment (RobertWarrenGilmore) [inline lib/model/graphParser.js]: * The angle bracket syntax is still nominally correct, but containers are not yet implemented, so the syntax is not used anywhere yet.
* Since there are not yet any containers, any test data that we have should have unique IDs.
* Global, because we don't have containers yet.

Comment (jboolean) [inline lib/model/graphParser.js]: This is not just reliant on containers. Think about IOs.

File: lib/model/node.js [hunk 25]
```diff
@@ -67,14 +67,14 @@ Node.prototype.isPinned = function() {
 /**
  * @method getIns - Getter method for the inputs property.
```
Comment (RobertWarrenGilmore) [inline lib/model/node.js]: Change the comment to match the actual method name.

File: lib/model/node.js [hunk 26]
```diff
@@ -67,14 +67,14 @@ Node.prototype.isPinned = function() {
...
  */
-Node.prototype.getIns = function() {
+Node.prototype.getInputs = function() {
   return this._inputs;
 };
 
 /**
  * @method getOuts - Getter method for the outputs property.
```
Comment (RobertWarrenGilmore) [inline lib/model/node.js]: method name

File: lib/model/node.js [hunk 27]
```diff
@@ -50,7 +50,7 @@ Node.prototype.getPos = function() {
  * @method setPos - Setter method for setting an xy-coordinate pair representing the Node's
  *         position in 2D space.
  */
-Node.prototype.setPos = function(x,y) {
+Node.prototype.setPos = function(x, y) {
```
Comment (RobertWarrenGilmore) [inline lib/model/node.js]: While you're at it, it would be cool if this were setPos**ition**.

Comment (RobertWarrenGilmore) [inline lib/model/node.js]: To match the method name in Port, that is.

File: lib/model/node.js [hunk 28]
```diff
@@ -42,7 +42,7 @@ Node.prototype.getID = function() {
 Node.prototype.getPos = function() {
   return{
     x: this._x,
-    y: this._y
+    y: this._y
```
Comment (RobertWarrenGilmore) [inline lib/model/node.js]: What changed?

File: lib/model/node.js [hunk 29]
```diff
@@ -29,9 +29,9 @@ var Node = function(id, inputs, outputs, styles, metadata) {
 };
 
 /**
- * @method getID - Getter method for the id property.
+ * @method getId - Getter method for the id property.
  */
-Node.prototype.getID = function() {
+Node.prototype.getId = function() {
```
Comment (RobertWarrenGilmore) [inline lib/model/node.js]: You could add `getEgo` and `getSuperEgo`, too. :stuck_out_tongue_winking_eye:

File: lib/utilities/positionUtils.js [hunk 30]
```diff
@@ -0,0 +1,114 @@
...
+var CardinalPortPosition = require('../model/cardinalPortPosition');
+var CartesianPortPosition = require('../model/cartesianPortPosition');
+/**
+ * @class PositionUtils
+ * @static
+ */
+var PositionUtils = {};
+
+// Data model scales values up to 0-100.
+var DATA_MODEL_MULTIPLIER = 100.0;
```
Comment (RobertWarrenGilmore) [inline lib/utilities/positionUtils.js]: I don't think that you need to add the `.0` to the end. JS has only one number type - `int` and `float` are not distinct.

File: lib/viewcomponents/app.jsx [hunk 31]
```diff
@@ -10,9 +10,9 @@ var App = React.createClass({
         return (
             <div id='app'>
                 <Graph model={this.props.model} />
-                <div id='model_editor'>
-                    <textarea onChange={this._onModelChange} value={JSON.stringify(this.props.model.getContents(), null, '  ')} />
-                </div>
+                {/*<div id='model_editor'>
+                    //<textarea onChange={this._onModelChange} value={JSON.stringify(this.props.model.getContents(), null, '  ')} />
+                //</div>*/}
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/app.jsx]: Why have you commented this out?

File: lib/viewcomponents/graph.jsx [hunk 32]
```diff
@@ -17,18 +17,14 @@ var Graph = React.createClass({
     },
 
     render: function() {
-        var data = this.props.model.getContents();
+        var graph = this.props.model.getGraph();
 
          // For each element of data.nodes, create a node component.
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/graph.jsx]: The comment doesn't match the variable names.

File: lib/viewcomponents/graph.jsx [hunk 33]
```diff
@@ -17,18 +17,14 @@ var Graph = React.createClass({
     },
 
     render: function() {
-        var data = this.props.model.getContents();
+        var graph = this.props.model.getGraph();
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/graph.jsx]: If you replaced the only usage of the old method, then why did you only deprecate it, rather than delete it?

File: lib/viewcomponents/node.jsx [hunk 34]
```diff
@@ -1,31 +1,109 @@
...
+      </g>
+    );
+  },
+
+  _getRenderedIOs: function(ioModels, isInput) {
+    return ioModels.map(function(ioModel, id) {
+      var position = this._getIOPosition(ioModel);
+
+      var onIOMove = function(event) {
+        this._onIOMove(event, ioModel);
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/node.jsx]: I don't see this function anywhere.

File: lib/viewcomponents/node.jsx [hunk 35]
```diff
@@ -1,31 +1,109 @@
...
+            {this._getRenderedIOs(model.getOutputs(), false)}
+      </g>
+    );
+  },
+
+  _getRenderedIOs: function(ioModels, isInput) {
+    return ioModels.map(function(ioModel, id) {
+      var position = this._getIOPosition(ioModel);
+
+      var onIOMove = function(event) {
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/node.jsx]: Where is this used?

File: lib/model/graphParser.js (hunk 23 above)
Comment (RobertWarrenGilmore) [inline lib/model/graphParser.js]: I see. IOs/Ports don't use the `>`. They use `+` and `-`. So there could be "Alpha+myInput" and "Bravo+myInput".

Comment (RobertWarrenGilmore) [inline lib/model/graphParser.js]: So if you need globally unique IDs, use the full addresses of the elements in question.

File: lib/model/node.js (hunk 28 above)
Comment (jboolean) [inline lib/model/node.js]: No idea.

File: lib/model/node.js (hunk 27 above)
Comment (jboolean) [inline lib/model/node.js]: Agreed. Executing…

File: lib/model/node.js (hunk 25 above)
Comment (jboolean) [inline lib/model/node.js]: Oversight. Fixing.

File: lib/model/node.js (hunk 26 above)
Comment (jboolean) [inline lib/model/node.js]: √

File: lib/test-data.json (hunk 12 above)
Comment (jboolean) [inline lib/test-data.json]: Labels being removed.

File: lib/utilities/positionUtils.js (hunk 30 above)
Comment (jboolean) [inline lib/utilities/positionUtils.js]: True. Changing.

File: lib/viewcomponents/app.jsx (hunk 31 above)
Comment (jboolean) [inline lib/viewcomponents/app.jsx]: Because we're now using data model objects not pure JSON objects throughout the system, the model cannot be directly displayed or edited anymore.

File: lib/viewcomponents/graph.jsx (hunk 33 above)
Comment (jboolean) [inline lib/viewcomponents/graph.jsx]: I'll delete it.

File: lib/viewcomponents/graph.jsx (hunk 32 above)
Comment (jboolean) [inline lib/viewcomponents/graph.jsx]: Correcting. Thanks for catching.

File: lib/viewcomponents/node.jsx (hunk 35 above)
Comment (jboolean) [inline lib/viewcomponents/node.jsx]: woops sorry

File: examples/public/stylesheets/main.less (hunk 2 above)
Comment (RobertWarrenGilmore) [inline examples/public/stylesheets/main.less]: Let's keep your change. Morgan can add this back in later if it's needed again.

File: lib/viewcomponents/app.jsx (hunk 31 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/app.jsx]: Damn. Okay. I hope we get the parser working soon, so that we can add back the editor panel, but with GCL in place of JSON.

**Comment by RobertWarrenGilmore** [comment]
I am okay with this PR. I give my vote to go ahead with the merge.
